jsk_recognition_utils/src/geo/convex_polygon.cpp

```diff
@@ -248,7 +248,10 @@ namespace jsk_recognition_utils
     Eigen::Vector3f c = centroid();
     Vertices new_vertices(vertices_.size());
     for (size_t i = 0; i < vertices_.size(); i++) {
-      new_vertices[i] = (vertices_[i] - c).normalized() * distance + vertices_[i];
+      Eigen::Vector3f diff_vec;
+      if (distance > 0.0) diff_vec = vertices_[i] - c;
+      else                diff_vec = c - vertices_[i];
```

Comment:
@wkentaro Please read the code, if `distance` is negative, we need to compute a vector of opposite direction.